server.go

```diff
@@ -25,6 +25,7 @@ type Server struct {
 	MaxIdleSeconds    int
 	MaxMessageBytes   int
 	AllowInsecureAuth bool
+	LenientSMTP       bool
```

Comment:
The default behaviour should probably to be lenient (since `RCPT TO` is lenient right now). So maybe we can change this to be a "Strict" flag instead?

Reply:
I didn't want to check existing behaviour, but yes, my first pass had StrictSMTP - I had to raid the antonyms page to get the most neutral opposite I could find. But happy to switch to StrictSMTP, sure.